sklearn/mixture/gaussianmixture.py

```diff
@@ -0,0 +1,685 @@
+                                                self.n_components, X.shape[1])
+        # check means
+        if self.init_means_ is not None:
+            self.init_means_ = _check_means(self.init_means_,
+                                            self.n_components, X.shape[1])
+        # check covars
+        if self.init_covars_ is not None:
+            self.init_covars_ = _check_covars(self.init_covars_,
+                                              self.n_components, X.shape[1],
+                                              self.covariance_type)
+
+        max_log_likelihood = -np.infty
+
+        if self.verbose > 0:
+            print_('The estimation of %s started.' %
+                   self.__class__.__name__, end='')
+
+        for init in range(self.n_init):
+            if self.verbose > 0:
+                print_('\nInitialization %s' % (init + 1), end='')
+                start_init_time = time()
+
+            self._initialize(X, self.init_weights_, self.init_means_,
+                             self.init_covars_)
+
+            current_log_likelihood = -np.infty
+            if self.verbose > 1:
+                print_('\n\tUsed %.5fs' % (time() - start_init_time), end='')
+
+            self.converged_ = False
+
+            for i in range(self.n_iter):
+                if self.verbose > 0:
+                    start_iter_time = time()
+
+                # e step
+                responsibilities = self._estimate_responsibilities(X)
+
+                # log_likelihood
+                prev_log_likelihood = current_log_likelihood
+                current_log_likelihood = self.score(X)
+
+                if self.verbose > 1:
+                    print_('\tLog-likelihood %s' % current_log_likelihood,
+                           end=' ')
+
+                change = abs(current_log_likelihood - prev_log_likelihood)
+                if change < self.tol:
+                    self.converged_ = True
+                    break
+
+                # m step
+                nk, xk, Sk = _sufficient_statistics(responsibilities, X,
+                                                    self.min_covar,
+                                                    self.covariance_type)
+                self._m_step(responsibilities, nk, xk, Sk)
+
+                if self.verbose > 0:
+                    print_('\n\tIteration %s' % (i + 1), end='')
+                if self.verbose > 1:
+                    print_('\tused %.5fs' % (time() - start_init_time),
+                           end=' ')
+            if not self.converged_ and self.verbose > 1:
+                current_log_likelihood = self.score(X)
+                print_('\tLog-likelihood %s' % current_log_likelihood, end='')
+
+            if self.verbose > 1:
+                print_('\n\tInitialization %s used %.5fs' %
+                       (init + 1, time() - start_init_time), end='')
+            if self.verbose > 0:
+                print
```

Comment:
Is this print really necessary ? In case it is, I guess it should be replaced by print_() or something. At the moment, the behaviour is going to be different between python 2 (print a new line) and 3 (do nothing).